File: week3-raindrops/ruby/raindrops.rb
=begin
Write your code for the 'Raindrops' exercise in this file. Make the tests in
`raindrops_test.rb` pass.

To get started with TDD, see the `README.md` file in your
`ruby/raindrops` directory.
=end

module Raindrops
  def self.convert(number)
    sound = ""

    if number % 3 == 0
      sound = "Pling"
    end

    if number % 5 == 0
      sound += "Plang"
    end

    if number % 7 == 0
      sound += "Plong"
    end

    if sound == ""
      return number.to_s
    end

    sound
  end
end

File: week3-raindrops/r/raindrops.R
raindrops <- function(number) {
  sound <- ""
  
  if (number %% 3 == 0) {
    sound <- "Pling"
  }
  if (number %% 5 == 0) {
    sound <- paste(sound, "Plang", sep = "")
  }
  if (number %% 7 == 0) {
    sound <- paste(sound, "Plong", sep = "")
  }

  if (sound == "") {
    return (as.character(number))
  }

  sound
}
